### ISSUE TITLE
Unable to Delete receipt after "Replace" is used by Admin

### ISSUE THREAD

**Issue by OfstadC** [Expensify/App#52345]
**Original GH:** https://github.com/Expensify/App/issues/50113#issuecomment-2468938344

## Action Performed:
1. From an employee account - navigate to the workspace chat and click + > Submit Expense > Scan Receipt > Complete submit flow
2. Login as admin 
3. Click on the receipt thumbnail
4. Click on "Replace" > Choose another receipt or take a photo of it > Upload
5. "Delete" > Confirm > Notice that an error is thrown
6. An error is shown: 'Unexpected...'

## Expected Result:

We should to two things here: 
- Enable replace functionality on backend so that admins and managers can replace receipts -- [Here's the logic](https://github.com/Expensify/Auth/blob/d91c55adbe30e496c7a1dcd05ea8af622a92dcb4/auth/lib/Receipt.cpp#L88-L90) that will need to be updated:
- Remove delete button for managers/admins, but keep the delete button for submitters -- this can be done as part of the [front-end issue](https://github.com/Expensify/App/issues/50113)

_______

## Screenshots/Videos
I didn't grab a new video when I reproduced this afternoon - but will grab one first thing tomorrow morning 

> jsonCode
> :
> 405
> message
> :
> "405 Cannot detach receipt from unowned transaction"
> onyxData
> :
> []
> requestID
> :
> "8e109b54d92daccc-MSP"

[Logs](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=request_id%3A%228e109b54d92daccc-MSP%22%20AND%20timestamp%3A%5B2024-11-11T00%3A00%20TO%202024-11-12T23%3A59%5D)

<details>
  <summary>Add any screenshot/video evidence</summary>

</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Julesssss</details>

**Comment by tgolen** [comment]
FYI there was also [this very small thread in Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1731002642067579) where we were talking about the expected behavior. So, if we are still questioning what the functionality should be, please respond in that thread.

**Comment by Julesssss** [comment]
The steps are correct. As discussed we want to allow admins to detach receipts **in NewDot**.  Though I have asked [here](https://expensify.slack.com/archives/C049HHMV9SM/p1731407829260699?thread_ts=1731002642.067579&cid=C049HHMV9SM) to clarify, as we may only want to support the replace case.

[Here's the logic](https://github.com/Expensify/Auth/blob/d91c55adbe30e496c7a1dcd05ea8af622a92dcb4/auth/lib/Receipt.cpp#L88-L90) that will need to be updated:

```
// Managers and admins cannot detach receipts unless they own the expense,
// but they can merge transactions within the same report.
if (!receiptID && !isManagerOrAdminMerging && (isManagerOrAdminAttaching && !canUserAttach)) {
    STHROW("405 Cannot detach receipt from unowned transaction");
}

```

**Comment by Julesssss** [comment]
Expected behaviour verified, we should to two things here:

- Enable replace functionality on backend so that admins and managers can replace receipts -- [Here's the logic](https://github.com/Expensify/Auth/blob/d91c55adbe30e496c7a1dcd05ea8af622a92dcb4/auth/lib/Receipt.cpp#L88-L90) that will need to be updated:
- Remove delete button for managers/admins, but keep the delete button for submitters -- this can be done as part of the https://github.com/Expensify/App/issues/50113

_I have updated the issue description_

**Comment by garrettmknight** [comment]
@Julesssss if you're gonna unassign can you make sure a BZ member is on it in the future?

**Comment by Julesssss** [comment]
Ah sure. I thought that was only necessary for external issues? We already have an external issue https://github.com/Expensify/App/issues/50113 for the front-end changes. LMK if I'm mistaken though

**Comment by joekaufmanexpensify** [comment]
Yeah, I am also curious about this. If there is nothing contributor-related to do here because of the separate FE issue, I feel like a BZ might not be needed. But also not sure if there is a defined process for this. Curious to hear what @garrettmknight thinks as well

**Comment by Julesssss** [comment]
I unassigned because unless I am mistaken; A) Engineers must triage but not necessarily implement auto-assigned engineering issues (including quality), and B) this is lower priority than other k2 issues, which include quality and migrate issues that will probably keep me busy until the end of the year.

Agree with this being a weird case, Matt Allen and I discussed this a bit [here](https://github.com/Expensify/App/issues/49162#issuecomment-2417632903)https://github.com/Expensify/App/issues/49162#issuecomment-2417632903.

**Comment by joekaufmanexpensify** [comment]
I also am a bit confused why we wouldn't just manage the backend portion of this bug in the original issue? This is also just an app issue. Seems more efficient to handle both in one issue (if they're both app issues), rather than work on two halves of the same bug separately.

**Comment by joekaufmanexpensify** [comment]
Looking into this more as I am a bit unsure if there's anything I should be doing here.

**Comment by Julesssss** [comment]
This is going to remain blocked until I can complete Deprecate Hybrid and some other WIP issues.

**Comment by Julesssss** [comment]
Unable to prioritise this over higher prioritised issues.

**Comment by Julesssss** [comment]
I'm working on external/internal issues that are going to take a while. Just noting for future reference that this is unlikely to be picked up until 2025

**Comment by twisterdotcom** [comment]
I am removing hot pick from any issues with an internal engineer assigned so other engineers can know it's available.

**Comment by Julesssss** [comment]
Same as above. I might share this with Garrett to list as an #expense issue that is looking for a volunteer

**Comment by Julesssss** [comment]
I shared in the #expense channel [here](https://expensify.slack.com/archives/C06ML6X0W9L/p1733913980171909).

**Comment by joekaufmanexpensify** [comment]
TY!

**Comment by Julesssss** [comment]
Awaiting volunteers. @garrettmknight if you don't mind, could you please highlight this backend issue in your next roundup? We just need someone with no assigned issue to implement [this](https://github.com/Expensify/App/issues/52345#issuecomment-2470178234) new logic.

>  we want to allow admins to detach receipts in NewDot

**Comment by joekaufmanexpensify** [comment]
Moving this one to weekly for now as I am OOO after today through new years and I see you are as well!

**Comment by joekaufmanexpensify** [comment]
OOO this week, we will circle back next week

**Comment by Julesssss** [comment]
We're seeking a volunteer for this one internally

**Comment by Julesssss** [comment]
No volunteer yet. Will ask again next week. I have higher priority issues to work on currently.

**Comment by garrettmknight** [comment]
@Julesssss dropping you off in case anyone is looking for Hot Picks and thinks you're handling.

**Comment by joekaufmanexpensify** [comment]
Pending internal volunteer

**Comment by joekaufmanexpensify** [comment]
[Posted again](https://expensify.slack.com/archives/C06ML6X0W9L/p1737569985597479) looking for a volunteer.

**Comment by joekaufmanexpensify** [comment]
Still pending an internal volunteer

**Comment by rlinoz** [comment]
Just assigned myself to this one

**Comment by joekaufmanexpensify** [comment]
awesome, thanks Rodrigo!

**Comment by rlinoz** [comment]
Going by the issue description

> We should to two things here: 
> - Enable replace functionality on backend so that admins and managers can replace receipts -- [Here's the logic]> (https://github.com/Expensify/Auth/blob/d91c55adbe30e496c7a1dcd05ea8af622a92dcb4/auth/lib/Receipt.cpp#L88-L90) that will need to be updated:
> - Remove delete button for managers/admins, but keep the delete button for submitters -- this can be done as part of the [front-end issue](https://github.com/Expensify/App/issues/50113)

I think we are already doing the first, we allow replacing but we don't allow deletion, but since step 2 is remove the delete button we should be good by just doing step 2, does that sound correct or am I missing something @joekaufmanexpensify @Julesssss ?

https://github.com/user-attachments/assets/ed56f8fe-078f-4d21-8dbc-07c27456b2c4

**Comment by joekaufmanexpensify** [comment]
Interesting. When this was first opened, it [sounded like](https://github.com/Expensify/App/issues/52345#issuecomment-2470952470) we thought we needed to make backend changes. Are you thinking there are actually no BE changes needed at all @rlinoz ?

**Comment by Julesssss** [comment]
I think at the time, [this](https://github.com/Expensify/App/issues/52345#issuecomment-2470178234) was the conversion we had about backend changes. It could well be different now? Or something else has changed.

**Comment by rlinoz** [comment]
Thanks, yeah, from the looks of it we want to have only the replace function and that is already working, so I think this will be only FE work which can be handler here https://github.com/Expensify/App/issues/50113

I will comment on that issue to let them know the next steps

**Comment by joekaufmanexpensify** [comment]
Ah, awesome. Does that mean we're good to close this one then?

**Comment by rlinoz** [comment]
I will wait for a reply in the other issue, but we probably are

**Comment by joekaufmanexpensify** [comment]
Sounds good, TY!

**Comment by joekaufmanexpensify** [comment]
I see we've moved forward with the FE in the other issue, so going to close this one out. Thanks again @rlinoz !
